python/pyiceberg/io/pyarrow.py

```diff
@@ -93,11 +88,23 @@ class PyArrowFile(InputFile, OutputFile):
         >>> # output_file.create().write(b'foobytes')
     """
 
-    def __init__(self, location: str, path: str, fs: FileSystem):
+    @staticmethod
+    def normalize_path(location: str) -> str:
```

Comment:
I don't think this class needs to change at all. The file system is already parsing the URI and can just provide the path like before.

I think you could probably argue that this class could parse the path out of the `location` in `__init__` if the path isn't provided, but I think that's minor.

Reply:
I've reverted most of the changes